File: app/src/main/java/com/allocate/ontime/encryption/EDHelper.java
package com.allocate.ontime.encryption;

//import com.allocate.ontime.retrofit.EDModel;
import com.google.gson.Gson;

import retrofit2.Response;
//import timber.log.Timber;

/**
 * Created by admin on 02-Feb-18.
 */

public class EDHelper {

    private static final String TAG = EDHelper.class.getSimpleName();

    public static final String KEY = "MDEyMzQ1Njc4OTAxMjM0NTY3ODkwMTIzNDU2Nzg5MzA=";


    public static EDModel encrypt(Object object) {
        String normalTextEnc = "";
        try {
            String temp = new Gson().toJson(object);
            temp = temp.replace("QEA\\u003d", "QEA=");
            normalTextEnc = AESHelper.getInstance().encrypt(KEY, temp);
        } catch (Exception e) {
            //LogFiles.appendEncryptionError("Error in encryption\nError : " + e.toString());
        }
        return new EDModel(normalTextEnc);
    }

    public static EDModel encryptString(String value) {
        String normalTextEnc = "";
        try {
            String temp = value.replace("QEA\\u003d", "QEA=");
            normalTextEnc = AESHelper.getInstance().encrypt(KEY, temp);
        } catch (Exception e) {
            //LogFiles.appendEncryptionError("Error in encryption\nError : " + e.toString());
        }
        return new EDModel(normalTextEnc);
    }
    public static <T extends Object> T decryptModel(Object o, Class<T> objClass) {
        String encrypted = "";

        String decrypted = "";
        try {
            encrypted = ((EDModel) ((Response) o).body()).getData();
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted);
        } catch (Exception e) {
            //LogFiles.appendEncryptionError("Error in decryption\nError : " + e.toString());
        }
        return new Gson().fromJson(decrypted, objClass);
    }

    public static <T extends Object> T decrypt(String encrypted, Class<T> objClass) {
        String decrypted = "";
        try {
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted);
        } catch (Exception e) {
            //LogFiles.appendEncryptionError("Error in decryption\nError : " + e.toString());
        }
        return new Gson().fromJson(decrypted, objClass);
    }
    public static String decrypt(String encrypted) {
        String decrypted = "";
        try {
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted);
        } catch (Exception e) {
//            Timber.tag("EDHelperError").e(e);
            //LogFiles.appendEncryptionError("Error in decryption\nError : " + e.toString());
        }
        return decrypted;
    }
}

File: app/src/main/java/com/allocate/ontime/encryption/EDHelper.kt
package com.allocate.ontime.encryption

import android.util.Log
import com.allocate.ontime.BuildConfig
import com.google.gson.Gson
import retrofit2.Response

/**
 * Created by Sublimity on 05/03/2024.
 */

object EDHelper {
    val TAG = "EDHelper"
    const val KEY = BuildConfig.KEY
    fun encrypt(`object`: Any?): EDModel {
        var normalTextEnc: String? = ""
        try {
            var temp = Gson().toJson(`object`)
            temp = temp.replace("QEA\\u003d", "QEA=")
            normalTextEnc = AESHelper.getInstance().encrypt(KEY, temp)
        } catch (e: Exception) {
            Log.d(TAG, "Error in encryption\\nError : \" + $e")
        }
        return EDModel(normalTextEnc!!)
    }

    fun encryptString(value: String): EDModel {
        var normalTextEnc: String? = ""
        try {
            val temp = value.replace("QEA\\u003d", "QEA=")
            normalTextEnc = AESHelper.getInstance().encrypt(KEY, temp)
        } catch (e: Exception) {
            Log.d(TAG, "Error in encryption\\nError : \" + $e")
        }
        return EDModel(normalTextEnc!!)
    }

    fun <T : Any?> decryptModel(o: Any, objClass: Class<T>?): T {
        var encrypted = ""
        var decrypted: String? = ""
        try {
            encrypted = ((o as Response<*>).body() as EDModel?)!!.data
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted)
        } catch (e: Exception) {
            Log.d(TAG, "Error in encryption\\nError : \" + $e")
        }
        return Gson().fromJson(decrypted, objClass)
    }

    fun <T : Any?> decrypt(encrypted: String?, objClass: Class<T>?): T {
        var decrypted: String? = ""
        try {
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted)
        } catch (e: Exception) {
            Log.d(TAG, "Error in encryption\\nError : \" + $e")
        }
        return Gson().fromJson(decrypted, objClass)
    }

    fun decrypt(encrypted: String?): String {
        var decrypted = ""
        try {
            decrypted = AESHelper.getInstance().decrypt(KEY, encrypted)
        } catch (e: Exception) {
            Log.d(TAG, "Error in encryption\\nError : \" + $e")
        }
        return decrypted
    }
}
